### ISSUE TITLE
[critical] build tests fail - references

### ISSUE THREAD

**Issue by ducin** [json-schema-faker/json-schema-faker#68]
Today I made one commit - changed `pateketrueke` into `json-schema-faker` in all markdown/readme links (https://github.com/json-schema-faker/json-schema-faker/commit/7b058b4b538613d40c8dc01077168ecc344ebf98). This caused the tests to fail...

The bug had to be somewhere there before - or the dependencies started to work in different way, dunno yet. Anyway, there are two similar fails

(1) I made that commit locally and my travis failed: https://travis-ci.org/ducin/json-schema-faker/builds/78351520

```
Failures:
  1) schema references (core/references.json) should handle internal $refs
   Message:
     schema references (Value 24 is not a multiple of 5)
```

(2) I moved the commit to official repo (travis hasn't failed yet and I ddin't expect that to!): https://travis-ci.org/json-schema-faker/json-schema-faker/builds/78351806

```
Failures:
  1) schema references (core/references.json) should handle internal $refs
   Message:
     schema references (Value 76 is not a multiple of 5)
```

it's the same test, but the generated value seems to be different in both cases. It seems like `multipleOf` is not interpreted correctly, somehow.

Dunno if that's relevant, but I can see that `deref` repo is failing as well: https://travis-ci.org/json-schema-faker/deref

@pateketrueke do you know something about what might have caused this `multipleOf` bug?

**Comment by ducin** [comment]
Once this is fixed, we may prepare 0.2.1 release and promote both the logo and the new version. But can't do it as long as the tests fail ;P

**Comment by pateketrueke** [comment]
Yes, it's a know issue, sometimes the generated numbers are wrong... my bad.

**Comment by ducin** [comment]
that's ok ;)  are you gonna fix that or do you need somebody's help?

**Comment by pateketrueke** [comment]
For numbers I need a little help, it's a weird behavior and needs more testing.

Actually I'm working on `deref` and `refaker` for solving `$ref` related issues, both seems stable on their develop branches, but still failing with jsf.

**Comment by ducin** [comment]
Do you have any clue on where should I look at? Is this bug somehow related to deref/refaker?

**Comment by pateketrueke** [comment]
In `develop` the specs still failing after dereferencing on numbers, may the new merge strategy break this?

I'm not sure since the number specs are now looking good, may be we need to add more.

If you can reproduce the same without using references would be nice feedback.

> Is this bug somehow related to deref/refaker?

Actually `jsf` performs a full dereferencing before faking anything so probably: yes.

I think we must think about a better strategy for lazy-dereferencing while traversing the schemas?

**Comment by ducin** [comment]
I've fixed one bug related to numbers (#70). I've also tried to bump up jsf:deref 0.3.0 to 0.4.0, but it fails two tests, see https://github.com/json-schema-faker/deref/issues/7.

**Comment by ducin** [comment]
Fixed, as the `0.2.1` release is passing on travis.
